docs/appnotes/0016-authorisation-in-tams-workflows.md

```diff
@@ -0,0 +1,324 @@
+| `/flows/{flowId}/avg_bit_rate`       | `HEAD`/`GET` | Request must have read permissions on {flowId}. Otherwise reject.                       |
+|                                      | `PUT`        | Request must have write permissions on {flowId}. Otherwise reject.                      |
+|                                      | `DELETE`     | Request must have write permissions on {flowId}. Otherwise reject.                      |
+| `/flows/{flowId}/segments`           | `HEAD`/`GET` | Request must have read permissions on {flowId}. Otherwise reject.                       |
+|                                      | `POST`       | Request must have write permissions on {flowId}, and either this must be the first registration of the object(s) (i.e. `/objects/{objectId}` returns 404) or the request must have read access to the object(s) being written. Otherwise reject.    |
+|                                      | `DELETE`     | Request must have write permissions on {flowId}. Otherwise reject.                      |
+| `/flows/{flowId}/storage`            | `POST`       | Request must have write permissions on {flowId}. Otherwise reject.                      |
+| `/objects/{objectId}`                | `HEAD`/`GET` | Restrict returned data in `referenced_by_flows` property by adding list of claimed auth classes to `flow_tag.auth_classes`. If the incoming request has `flow_tag.auth_classes` set, the request must be processed with `flow_tag.auth_classes` set to the intersection of the claimed auth classes and the provided list in `flow_tag.auth_classes`. |
+| `/flow-delete-requests`              | `HEAD`/`GET` | Request must have admin permissions. Otherwise reject.                                  |
+| `/flow-delete-requests/{request-id}` | `HEAD`/`GET` | Request must have delete permissions on the Delete Request's Flow ID. Otherwise reject. |
+
+### Determining base permissions
+
+#### Flows
+
+Read, write, and delete permissions on individual flows may be determined via auth classes listed in the `auth_classes` tag on the flow.
+This may be done via the `/flows/{flowId}/tags/auth_classes` endpoint.
+
+#### Sources
+
+Read, write, and delete permissions on individual sources may be determined via auth classes listed in the `auth_classes` tag on the source.
+This may be done via the `/sources/{sourceId}/tags/auth_classes` endpoint.
+
+#### Objects
+
+Read, write, and delete permissions on individual objects may be determined by filtering returned flows on the object.
+This may be done by setting `flow_tag.auth_classes` to relevant claimed auth classes (e.g. auth classes with read permissions if read permissions on the object are to be verified).
+If `referenced_by_flows` in the returned data is empty, the request DOES NOT have the relevant permissions.
+If `referenced_by_flows` in the returned data is empty, the request DOES have the relevant permissions.
```

Comment:
Should this be "not empty"? At present the condition is the same as the line above so is contradictory.